azure/scope/managedcontrolplane.go

```diff
@@ -402,7 +402,11 @@ func (s *ManagedControlPlaneScope) IsVnetManaged() bool {
 	ctx := context.Background()
 	ctx, log, done := tele.StartSpanWithLogger(ctx, "scope.ManagedControlPlaneScope.IsVnetManaged")
 	defer done()
-	isManaged, err := virtualnetworks.New(s).IsManaged(ctx)
+	virtualNetworksSvc, err := virtualnetworks.New(s)
+	if err != nil {
+		log.Error(err, "failed to create virtualnetworks service")
+	}
+	isManaged, err := virtualNetworksSvc.IsManaged(ctx)
```

Comment:
`IsManaged` looks like it will panic when `virtualNetworksSvc` is nil, so I think we might need to return early if `virtualnetworks.New` returns an error. I'm not sure what the most reasonable bool to return would be though.

Reply:
It seems like the effective default is `false`, so I changed it to log and return that.

It would be great to return an error here, but this func is called from a hundred different places.